src/txn/txn_timestamp.c

```diff
@@ -902,46 +941,53 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     __wt_config_init(session, &cparser, cfg[1]);
     while ((ret = __wt_config_next(&cparser, &ckey, &cval)) == 0) {
         WT_ASSERT(session, ckey.str != NULL);
-        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len)) {
+        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "commit", &commit_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "durable", &durable_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("prepare_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("prepare_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "prepare", &prepare_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("read_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("read_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "read", &read_ts, &cval));
-            set_ts = true;
-        }
     }
     WT_RET_NOTFOUND_OK(ret);
 
-    /* Look for a commit timestamp. */
-    if (commit_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_commit_timestamp(session, commit_ts));
-
-    /*
-     * Look for a durable timestamp. Durable timestamp should be set only after setting the commit
-     * timestamp.
-     */
-    if (durable_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_durable_timestamp(session, durable_ts));
-    __wt_txn_publish_durable_timestamp(session);
+    return __wt_txn_set_timestamp_internal(session, commit_ts, durable_ts, prepare_ts, read_ts);
+}
```

Comment:
Missing `()` around return call.

Reply:
Thanks, I'm surprised `s_all` doesn't catch this - I'll fiddle with it today and see if I can easily make that work, otherwise I'll file an issue.

Reply:
I suspect the problem is that `s_all` doesn't have a real C parser, and so "we should return xxx instead of yyy" in a comment will trigger any simple test.

Reply:
I had a play with it, I think it'll be easy enough with the `--word-regexp` option we're using for other checks. It's out of scope for this change though, I'll make a ticket later today. In the meantime, this is fixed in 249df1c.